ocaml/testsuite/tests/typing-modes/val_modalities.ml

```diff
@@ -549,3 +549,63 @@ module type S' =
 
 (* CR zqian: add tests of recursive modules & include w/ modalties, once
    modules can have modes. *)
+
```

Comment:
It might be nice to see how this feature interacts with `include`, not just `include functor`. Also it would be good to see that the `@@ portable` thing in a `sig` is disregarded in the module inclusion check between two module types -- one with `@@ portable` up top and the other with it per-function.